### ISSUE TITLE
driftctl hangs.

### ISSUE THREAD

**Issue by sojjan1337** [snyk/driftctl#1653]
Hi!

When i run a scan againt s3, driftctl hangs. And i do not get any info from debug:

`Scan duration: 52s
Provider version used to scan: . Use --tf-provider-version to use another version.`

But when i run a scan locally it doesn't hang, and the debug shows me the following:

`can duration: 11s
Provider version used to scan: . Use --tf-provider-version to use another version.
DEBU[0013] Closing providers
DEBU[0013] Closing provider key=aws
DEBU[0013] Closing gRPC client alias=eu-north-1`

**Comment by thesagarmatha** [comment]
Yep, I am having the same issue. Thanks for bringing up this issue.

**Comment by ramkalapala** [comment]
hi, the hang is in Cleanup() in enumeration/remote/terraform/provider.go:

func (p *TerraformProvider) Cleanup() {
    for alias, client := range p.grpcProviders {
        logrus.WithFields(logrus.Fields{
            "alias": alias,
        }).Debug("Closing gRPC client")
        client.Close()  // ← blocking, no timeout
    }
}

client.Close() is a bare blocking call with no timeout. The debug output confirms the scan completes fine and hangs exactly at "Closing gRPC client alias=eu-north-1" in slower/containerised environments where gRPC shutdown doesn't return cleanly.

Fix would be wrapping with a timeout:

func (p *TerraformProvider) Cleanup() {
    for alias, client := range p.grpcProviders {
        logrus.WithFields(logrus.Fields{"alias": alias}).Debug("Closing gRPC client")
        done := make(chan struct{})
        go func(c *plugin.GRPCProvider) {
            c.Close()
            close(done)
        }(client)
        select {
        case <-done:
        case <-time.After(5 * time.Second):
            logrus.WithField("alias", alias).Warn("gRPC client close timed out, forcing exit")
        }
    }
}
